yare-engine/src/main/java/com/sabre/oss/yare/engine/executor/DefaultRulesExecutor.java

```diff
@@ -184,6 +192,32 @@ private long getPriority(Rule a) {
         return attribute != null ? (Long) attribute.getValue() : 0L;
     }
 
+    private static class StopProcessingListener implements EngineListener {
+        private static final Logger log = LoggerFactory.getLogger(StopProcessingListener.class);
+
+        private final AtomicBoolean evaluationTerminated;
+
+        StopProcessingListener(EngineControllerObservable engineController) {
+            evaluationTerminated = new AtomicBoolean(false);
+            if (engineController != null) {
+                engineController.register(this);
+            }
+        }
+
+        boolean isEvaluationTerminated() {
+            return evaluationTerminated.get();
+        }
+
+        @Override
+        public void onEvent(EngineControllerEvent event) {
+            Objects.requireNonNull(event, "event cannot be null");
+            if (EngineControllerEventType.STOP_PROCESSING_ALL_FACTS.equals(event.getType())) {
```

Comment:
We can compare this enums with ==.

Reply:
Outdated, but for future can be helpful :)